HIP-0006.md

```diff
@@ -0,0 +1,170 @@
+the following format:
+
+```
+vin:
+  0: current name owner, encumbered by Swap Script
+  1: A UTXO controlled by the bidder of some value V
+vout:
+  0: TRANSFER to the bidder's address
+```
+
+Input 1 is signed with `SIGHASH_ANYONECANPAY | SIGHASH_SINGLEREVERSE`. This commits
+the incoming payment to the name transfer. Note that Input 0 is not signed yet,
+but it must be the current name-owning UTXO or the linked output 0 would be invalid.
+
+Note that this transaction has no change output. Because the bidder is signing
+with single-reverse, they can't specify a change output in addition to the name
+transfer output. If they were to sign with `SIGHASH_ALL` then the next step of
+the protocol would not be possible. If they signed with `SIGHASH_SINGLE` they could
+commit to a change address, but not the name transfer.
+
+Having no change output means the INPUT coin they are spending MUST BE THE VALUE
+OF THE BID THEY INTEND TO PLACE ON THE NAME. This is a big inconvenience
+that can be solved in at least these two ways:
+
+- The user prepares their wallet in advance and collects UTXOs with a range
+of values (1 HNS, 10 HNS, 100 HNS, 1000 HNS, etc).
+
+- When the user wants to bid, their wallet creates a "perfect" UTXO by spending
+to itself, creating a coin with the exact value desired. This output can be used
+(even before it is confirmed) in this protocol.
+
+### Auction FINALIZE
+
+The name owner collects partially-signed bids and after some time, declares the highest
+value bid the winner. The name owner completes the auction by adding a payment output
+to the partially-signed transaction, signing and broadcasting:
+
+```
+vin:
+  0: current name owner, encumbered by Swap Script
+  1: A UTXO controlled by the bidder of some value V
+vout:
+  0: TRANSFER to the bidder's address
+  1: payment to the name-owner
+```
+
+The name owner pays themselves from the fully funded transaction, and signs input 0
+with `SIGHASH_ALL`. This transaction is a valid atomic swap and can be broadcast to the network.
+Because the locking script prevents all other name actions, the auction winner can
+`FINALIZE` this name transfer any time after the 288-block lockout without any other
+obligation.
+
+Note that since the bidder creates the INPUT and the name owner creates the OUTPUT,
+the name owner MUST also pay the miner fee for this transaction. They do so by reducing
+the value of output 1, effectively paying the miner fee out of the money they earned
+by the sale.
+
+## Extra Notes
+
+### Bids must remain valid off-chain
+
+The partially-signed bid transactions are collected by the name owner and commit
+to spending a UTXO of some value V from the blockchain. In the simplest construction,
+nothing prevents the bidder from invalidating this off-chain transaction by
+double-spending the coin they used to bid. For this reason, the name owner must
+routinely check that all submitted bids are still valid. They can try finalizing
```

Comment:
This could also be another use case for an auction platform that might take care of the auction finality issue mentioned below. Not sure if important enough to highlight?

Reply:
Sure, will add.